R/MatrixGenerics-package.R

```diff
@@ -21,3 +21,84 @@ setClassUnion("matrix_OR_array_OR_table_OR_numeric",
     c("array", "table", "numeric")
 )
 
+
+## - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+## make_default_method_def()
+##
+
+## All packages listed below must also be listed in the Suggests field.
+## They are expected to implement methods for the generics defined in
+## MatrixGenerics. No need to list matrixStats here as it is special and
+## already imported by default.
+.SUGGESTED_PACKAGES_TO_SEARCH <- c(
+    "sparseMatrixStats",
+    ## We list DelayedMatrixStats even though the methods defined in it
+    ## won't be found by the generics in MatrixGenerics. This is because
+    ## DelayedMatrixStats defines its own matrixStats generics.
+    "DelayedMatrixStats"
+    # ... add more packages in the future
+)
+
+.long_and_fancy_errmsg <- function(crude_errmsg, unloaded_pkgs)
+{
+    plural <- length(unloaded_pkgs) > 1L
+    errmsg <- paste0(crude_errmsg,  "\n  However, the following package",
+        if (plural) "s are" else " is",
+        " likely to contain the missing method\n  but ",
+        if (plural) "are" else "is", " not installed: ",
+        paste0(unloaded_pkgs, collapse=", "), "\n  ",
+        "Please install ", if (plural) "them" else "it",
+        " (with 'BiocManager::install()') and try again.")
```

Comment:
Would it be possible to include the `unloaded_pkgs` in the `BiocManager::install()` call so a user could copy-paste it?